src/OpenIddict.Mvc/OpenIddictController.cs

```diff
@@ -38,7 +38,7 @@ public class OpenIddictController<TUser, TApplication> : Controller where TUser
         protected virtual OpenIddictOptions Options { get; }
 
         [HttpGet, HttpPost]
-        public async Task<IActionResult> Authorize() {
+        public virtual async Task<IActionResult> Authorize() {
```

Comment:
virtual add the option to subclass and override what you need and also call base. I like the virtual thing.